app/src/main/java/com/m/services/auth/LoginController.java

```diff
@@ -1,56 +1,54 @@
-package com.m.helper;
-
+package com.m.services.auth;
 
 import android.content.Context;
 import android.content.Intent;
 import android.net.Uri;
-import android.os.Bundle;
 
 import androidx.annotation.Nullable;
-import androidx.appcompat.app.AppCompatActivity;
 
-import com.m.helper.Auth.AuthStateManager;
-import com.m.androidNativeApp.MainActivity;
+import com.m.helper.MobileService;
 
 import net.openid.appauth.AuthState;
 import net.openid.appauth.AuthorizationException;
+import net.openid.appauth.AuthorizationRequest;
 import net.openid.appauth.AuthorizationResponse;
 import net.openid.appauth.AuthorizationService;
 import net.openid.appauth.AuthorizationServiceConfiguration;
-import net.openid.appauth.AuthorizationRequest;
 import net.openid.appauth.ResponseTypeValues;
 import net.openid.appauth.TokenRequest;
 import net.openid.appauth.TokenResponse;
 
-public class LoginActivity extends AppCompatActivity {
+import static com.m.services.auth.LoginActivity.RE_AUTH;
+import static com.m.services.auth.LoginActivity.mAuthStateManager;
 
-    public static AuthState mAuthState;
-    public static AuthStateManager mAuthStateManager;
-    public static MobileService mobileService;
+public class LoginController {
+    private Context context;
+    private MobileService mobileService;
     private AuthorizationService mAuthService;
-    private int RC_AUTH = 100;
-    public static int RE_AUTH = 0;
+    private LoginListener listener;
+    public static AuthState mAuthState;
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        mAuthStateManager = AuthStateManager.getInstance(this);
-        Context context = getApplicationContext();
-        mobileService = MobileService.getInstance(context.getApplicationContext());
 
-        if (mAuthStateManager.getCurrent().isAuthorized() && RE_AUTH == 0) {
-            System.out.println("User is authorized already");
-            startActivity(new Intent(this, MainActivity.class));
+    public LoginController(Context context){
+        this.context = context;
+        mobileService = MobileService.getInstance();
 
-            finish();
+    }
 
-        } else {
-            doAuth();
-        }
+    public LoginController(Context context, LoginListener listener, MobileService mobileService){
+        this.context = context;
+        this.listener = listener;
+        this.mobileService = mobileService;
+    }
+
+    public void reAuthorise() {
+        RE_AUTH = 403;
```

Comment:
why do we change the `RE_AUTH` here?

In Java, the convention is that is a variable is named like this (all uppercases), it usually means the value is a constant value and is read only, and shouldn't be changed at all.